### ISSUE TITLE
The parameter subscribed_fields is required

### ISSUE THREAD

**Issue by rafayet-monon** [jgorset/facebook-messenger#231]
I am facing this error with facebook-messenger-0.11.1. 

`raise_errors': (#100) The parameter subscribed_fields is required (Facebook::Messenger::Subscriptions::Error)

	from /home/user/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/facebook-messenger-0.11.1/lib/facebook/messenger/subscriptions.rb:20:in `subscribe'

How can I fix this error?

**Comment by amrutjadhav** [comment]
@rafayet-monon facebook-messenger-0.11.1 gem works on graph API v2.6 and as of July 13, 2018, this API is unavailable. So I would suggest you, use facebook-messenger-1.1.1 which works on graph API v2.9. And graph API v2.9 don't need `subscribed_fields` parameter while subscription.

**Comment by rafayet-monon** [comment]
Same here, @amrutjadhav . Are we missing anything. It would be a big help if you pointed out.

**Comment by amrutjadhav** [comment]
@rafayet-monon It's very strange. Because currently I am using facebook-messenger-v1.1.1 and everything is working fine for me. If you share your code/sample, I could check it out.

**Comment by rafayet-monon** [comment]
@amrutjadhav  Actually I am trying to run [this project with api-ai](https://github.com/duc4nh/wbot-api-ai). [here is the gist](https://gist.github.com/rafayet-monon/cad2650c967d77397a874e4057a9780c) that contains connection to subscription.

**Comment by amrutjadhav** [comment]
@rafayet-monon I couldn't understand the reason behind your error as the shared code looks correct. And documentation of subscription on Graph API v2.9 also don't require `subscription_fields`. I would suggest you to check FB developer panel and check whether everything is OK. You could also try by creating a new app and check if the error still exists.

**Comment by artyomlagun** [comment]
hello guys! 
I faced with a similar problem. 
my FB application created with API v3.2 and I assume this problem happened, because of  FB API v3.1+ require subscription_fields (https://developers.facebook.com/docs/graph-api/reference/page/subscribed_apps#Creating).

I think if add the second defined parameter to `subscribe` method, like 
```ruby
def subscribe(access_token:, subscription_fields: [])
```
it will fix problem

**Comment by artyomlagun** [comment]
but I could be wrong

**Comment by artyomlagun** [comment]
@jgorset  I think, it will be helpful https://github.com/jgorset/facebook-messenger/pull/232

**Comment by amrutjadhav** [comment]
@artyomlagun https://github.com/jgorset/facebook-messenger/pull/232#issuecomment-438509249

**Comment by Ericfreespirit** [comment]
Same problem guys :(

**Comment by Snakultime** [comment]
Same problem too ^^'
Facebook has change API 2.9 to 3.2. Maybe it's that.

**Comment by Fredis** [comment]
Guys,

Does anyone found a solution? I got the same error on Heroku: "(#100) The parameter subscribed_fields is required. (Facebook::Messenger::Subscriptions::Error)" ?

Best regards,
Fred.

**Comment by Snakultime** [comment]
OK!; Just remove 'Facebook::Messenger::Subscriptions.subscribe(access_token: access_token)'
And working good.

**Comment by jgorset** [comment]
This is happening because [this endpoint targets Graph API 2.9](https://github.com/jgorset/facebook-messenger/blob/master/lib/facebook/messenger/subscriptions.rb#L12), but new apps no longer support targeting that. We'll need to upgrade the endpoint to using Graph API 3.2, and impelement support for `subscription_fields` to the `[subscribe](https://github.com/jgorset/facebook-messenger/blob/master/lib/facebook/messenger/subscriptions.rb#L30)` method.

You were onto something, then, @artyomlagun – the only thing you didn't do was to upgrade the version in the endpoint! If you revise your pull request, I'll be happy to merge it.

*Here are the options for an app created a long time ago*

<img width="1086" alt="screen shot 2018-12-05 at 11 11 23" src="https://user-images.githubusercontent.com/195266/49485584-d81ac180-f87e-11e8-9afd-f89b1eb95ffd.png">

*Here are the options for an app created today*

<img width="656" alt="screen shot 2018-12-05 at 11 12 51" src="https://user-images.githubusercontent.com/195266/49485589-dc46df00-f87e-11e8-85cf-94df9da7c12b.png">

**Comment by joaotorres** [comment]
Any updates on this? Did anyone manage to make this work?

**Comment by joaotorres** [comment]
> @artyomlagun [#232 (comment)](https://github.com/jgorset/facebook-messenger/pull/232#issuecomment-438509249)

I used your gem, but seems you misnamed the parameter, or I don't know if it had another name on the previous version. I cloned it and changed the name (from `subscription_fields` to `subscribed_fields`), but then it still complains if the field is empty, though with another error:

```
Unable to load application: Facebook::Messenger::Subscriptions::Error: An unknown error has occurred.
```

But then if you specify one of the fields, it complains about permissions:

```
Unable to load application: Facebook::Messenger::Subscriptions::Error: (#200) To subscribe to the messages field, one of these permissions is needed: manage_pages
```

Any ideas?

**Comment by jgorset** [comment]
I took @artyomlagun's pull request the rest of the way and released a new version 1.1.2 that should fix this issue.

**Comment by habipyesilyurt17** [comment]
If you are using Graph API v3.2 or upper version, please add one additional parameter in the POST request ,

subscribed_fields=leadgen

**Comment by stabgan** [comment]
what's leadgen ?

**Comment by artyomlagun** [comment]
@habipyesilyurt17  @stabgan 

which params subscribed_apps use https://developers.facebook.com/docs/graph-api/reference/page/subscribed_apps#Creating - you able to see in the Parameters section (the leadgen on the 9th line from the bottom).

and you should call subscribe method like: 

`Facebook::Messenger::Subscriptions.subscribe(access_token: @page_token, subscribed_fields: SUBSCRIBED_PARAMS, page_id: @page_id)`

your SUBSCRIBED_PARAMS should be:
`
SUBSCRIBED_PARAMS = [
    'messages', 'message_echoes', 'message_deliveries', 'messaging_optins', 'leadgen'
  ]
`

**Comment by artyomlagun** [comment]
> I took @artyomlagun's pull request the rest of the way and released a new version 1.1.2 that should fix this issue.

sorry, I've not seen your comment :) but I'm glad that my PR was helpful
